### ISSUE TITLE
`no-invalid-regexp` when using the `d` flag

### ISSUE THREAD

**Issue by jespertheend** [denoland/deno_lint#985]
### Lint Name
`no-invalid-regexp`

### Code Snippet

```ts
/nooo/gd;
```

### Expected Result
No linting errors.

### Actual Result
Both `no-empty-character-class` and `no-invalid-regexp` are reported.

### Additional Info
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/hasIndices

### Version
deno 1.17.0 (release, x86_64-apple-darwin)
v8 9.7.106.15
typescript 4.5.2
